pkg/image/prune/prune.go

```diff
@@ -1190,3 +1160,62 @@ func getName(obj runtime.Object) string {
 	}
 	return fmt.Sprintf("%s/%s", accessor.GetNamespace(), accessor.GetName())
 }
+
+// tryProtocolsWithRegistryURL runs given callback with different protocols until no error is returned. The
+// https protocol is the first attempt. If it fails and allowInsecure is true, http will be the next. Obtained
+// errors will be concatenated and returned.
+func tryProtocolsWithRegistryURL(registry string, allowInsecure bool, callback func(registryURL url.URL) error) (*url.URL, error) {
```

Comment:
`callback` here suggests an actual callback method, but it's not. It's the heart of this method, an action performed by this function. I'd suggest renaming the variable.